src/storage/secondary/column/primitive_column_builder.rs

```diff
@@ -111,11 +124,32 @@ impl<T: PrimitiveFixedWidthEncode> ColumnBuilder<T::ArrayType> for PrimitiveColu
         while iter.peek().is_some() {
             if self.current_builder.is_none() {
                 if self.nullable {
-                    self.current_builder = Some(BlockBuilderImpl::PlainNullable(
-                        PlainPrimitiveNullableBlockBuilder::new(
-                            self.options.target_block_size - 16,
-                        ),
-                    ));
+                    if self.options.is_rle {
```

Comment:
Consider using `match` here, otherwise it's hard to read...

```rust
match (self.nullable, self.options.is_rle) {
  (true, true) => xxx,
  (true, false) => xxx,
  // ...
}
```